packages/contracts/contracts/discovery/Controller.sol

```diff
@@ -2,67 +2,82 @@
 pragma solidity =0.8.12;
 
 import {IController} from "./IController.sol";
-import {ProjectHelpers} from "../libraries/ProjectHelpers.sol";
+import {IProject} from "./IProject.sol";
+import {Project} from "./Project.sol";
+import {IBatch} from "./IBatch.sol";
+import {Batch} from "./Batch.sol";
 
-contract Controller is IController {
-    using ProjectHelpers for Project;
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
 
-    uint256 lastProjectId;
-    uint256 lastBatchId;
+contract Controller is IController, AccessControl {
+    mapping(address => bool) public projects;
+    mapping(address => address) public projectsToBatches;
 
-    mapping(uint256 => Project) public projects;
+    address public stakingContract;
 
-    mapping(uint256 => Batch) public batches;
+    event RegisterProject(address project);
 
-    /// Batch => user => votes
-    mapping(uint256 => mapping(address => uint256)) public userVoteCount;
-
-    /// Batch => projectId => votes
-    mapping(uint256 => mapping(uint256 => uint256)) public projectVoteCount;
-
-    event RegisterProject(uint256 projectId);
-
-    /// @inheritdoc IController
-    function getProject(uint256 id) external view returns (Project memory) {
-        return projects[id];
+    constructor() {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
     }
 
     /// @inheritdoc IController
-    function getBatch(uint256 id) external view returns (Batch memory) {
-        return batches[id];
+    function getBatch(address _project) external view returns (address) {
```

Comment:
maybe `getBatchForProject` is more clear. otherwise it looks a simpler getter for a batch